src/main/java/com/google/ehub/utility/UserRecommendationUtils.java

```diff
@@ -0,0 +1,53 @@
+package com.google.ehub.utility;
+
+import com.google.ehub.data.FavoriteItemDatastore;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Utility class that implements the algorithm used to recommend the users that have the most common
+ * number of shared likes.
+ */
+public final class UserRecommendationUtils {
+  private static final FavoriteItemDatastore favoriteItemDatastore =
+      FavoriteItemDatastore.getInstance();
+
+  /**
+   * Finds the most recommended emails up to a given limit.
+   *
+   * @param userEmail the email of the user that wants to search for its recommended email list
+   * @param maxNumberOfRecommendations the maximum desired size for the recommendation list
+   * @return list containing the most recommended emails
+   */
+  public static List<String> getRecommendedEmails(
+      String userEmail, int maxNumberOfRecommendations) {
+    List<Long> itemIdsLikedByUser = favoriteItemDatastore.queryFavoriteIds(userEmail);
+    List<List<String>> emailsThatLikedSameItem =
+        getEmailsThatLikeSameItemsAsUser(itemIdsLikedByUser);
+
+    return getRecommendedEmails(emailsThatLikedSameItem, maxNumberOfRecommendations);
+  }
+
+  private static List<String> getRecommendedEmails(
```

Comment:
would be good I think to make the 2 func names a little more different

Reply:
I changed the function name to ```getRecommendedEmailsFromEmailList()``` I couldn't figure out a better name, so let me know if you could have a better suggestion.